libs/MVS/Scene.cpp

```diff
@@ -1373,4 +1373,97 @@ bool Scene::ScaleImages(unsigned nMaxResolution, REAL scale, const String& folde
 	}
 	return true;
 } // ScaleImages
+
+/*----------------------------------------------------------------*/
+// detect region-of-interest based on camera positions, directions and sparse points
+// scale specifies the ratio of the ROI's diameter
+bool Scene::DetectROI(float scale)
+{
+    typedef CLISTDEF0IDX(Camera, uint32_t) CamArr;
+    CamArr camArr;
+    FOREACH(i, images) {
+        const Image &imageData = images[i];
+        if (!imageData.IsValid())
+            continue;
+        camArr.emplace_back(imageData.camera);
+    }
+    unsigned nCams = camArr.size();
+    if (nCams < 3) {
+        VERBOSE("error: not enough valid views for the ROI detection");
+        return false;
+    }
+    // compute the camera center and the direction median
+    typedef CLISTDEF0IDX(float, IDX) Scalars;
+    Scalars x(nCams), y(nCams), z(nCams), nx(nCams), ny(nCams), nz(nCams);
+    FOREACH(i, camArr) {
+        Point3f camC(camArr[i].C);
+        x[i] = camC.x;
+        y[i] = camC.y;
+        z[i] = camC.z;
+        Point3f camDirect(camArr[i].Direction());
+        nx[i] = camDirect.x;
+        ny[i] = camDirect.y;
+        nz[i] = camDirect.z;
+    }
+    CMatrix camCenter(x.GetMedian(), y.GetMedian(), z.GetMedian());
+    CMatrix camDirectMed(nx.GetMedian(), ny.GetMedian(), nz.GetMedian());
+    float camDirectMedLen = (float) ((const CMatrix ::EVec) camDirectMed).norm();
+    camDirectMed /= camDirectMedLen;
+    if (VERBOSITY_LEVEL > 2)
```

Comment:
pls surround with
```	#if TD_VERBOSE != TD_VERBOSE_OFF
	#endif
```

Reply:
Please help to confirm if it is correct, thx!

```cpp
    ...
    camDirectMed /= camDirectMedLen;
    #if TD_VERBOSE != TD_VERBOSE_OFF
    if (VERBOSITY_LEVEL > 2)
        VERBOSE("The camera positions median is (%f,%f,%f), directions median and norm is (%f,%f,%f), %f",
                camCenter.x, camCenter.y, camCenter.z, camDirectMed.x, camDirectMed.y, camDirectMed.z, camDirectMedLen);
    #endif
    ...
```

Reply:
this is fine, the reason for this is that verbosity can be disabled completly, and in that case VERBOSITY_LEVEL does not even exists anymore

Reply:
I see. Thx!